app/src/main/java/com/simplecity/amp_library/ui/fragments/AlbumFragment.java

```diff
@@ -129,7 +135,8 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
             layoutManager.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
                 @Override
                 public int getSpanSize(int position) {
```

Comment:
There's a class in Shuttle which I'm not actually using at the moment, which simplifies this.

See `com.simplecityapps.recycler_adapter.recyclerview.SpanSizeLookup`. If you attach this to the `LayoutManager`, then the span size is automatically gleaned from the `ViewModel.getSpanSize()` method - so we no longer have to manually implement the span count. 

So, we'd just have `layoutManager.setSpanSizeLookup(new SpanSizeLookup(spanCount));`

The default `spanSize` is `spanCount`, meaning if you have a grid with a span of '2', by default a `ViewModel` will span 2 columns. You just have to ensure that `getSpanSize()` has been overridden in any `ViewModel` which is supposed to only span one column (such as `AlbumView`).

If this sounds complicated, let me know and I'll make this change myself.

Reply:
Sounds simple enough, I will take a look at it.